### PR TITLE
Migrate selection for link checker

### PR DESCRIPTION
**Pull request by skoszuta**

### CODE REVIEW

File: tools/links-check/report.ts [hunk 1]
```diff
@@ -70,14 +72,15 @@ export function printReport(report: Report): Summary {
       console.info(`⚠️  Yet there are ${outdated} outdated links. See above.`);
     }
   } else {
-    console.info(`⁉️  Detected some potentially broken links ${broken.length}/${total}`);
+    console.info(`⁉️  Detected some potentially broken links ${broken.length}/${total}:`);
+    console.info();
     for (const { file, link } of broken) {
-      const ico = link.updated ? "⚠️" : "🦖";
-      console.info(`  ${ico} ${link.raw}`);
+      console.info(`  ⚠️  ${link.url}`);
```
[the remark this answers is not in the document]
Comment (skoszuta) [reply tools/links-check/report.ts:78]: Oki, restoring the dino. However note that from now on im showing links that link to existing blocks but failed migration as (most likely) broken.

File: tools/links-check/index.ts [hunk 2]
```diff
@@ -7,31 +12,93 @@ main().catch((err: unknown) => {
...
+  return null;
+}
+
 async function main() {
-  const files = process.argv.slice(2);
+  const args = process.argv.slice(2);
+
+  const ignoreFileIndex = args.indexOf("--ignore-file");
+  let ignorePatterns: string[] = [];
+
```
Comment (tomusdrw) [inline tools/links-check/index.ts:38]: Please extract support for CLI parameter `--ignore-file` (that would be parsed by `ignore` package), but is not expected to automatically find `.gitignore` files to a separate PR.

While it's a single CLI parameter, there is for sure going to be more, so we should think about better (more generic) way to extract these parameters from `args`.

File: tools/links-check/migrate.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import * as fs from "node:fs";
+import * as path from "node:path";
+import * as readline from "node:readline";
+import type { Report } from "./report";
+
+export async function promptUserForMigration(): Promise<boolean> {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
```
Comment (tomusdrw) [inline tools/links-check/migrate.ts:10]: Please move to a separate PR. The user should not be prompted for anything, we should rather provide a CLI flag to auto-overwrite the files. The tool is meant to be running in a non-interactive environment, just like linters, so on CI we can check that links are valid, and locally we can migrate the hooks by running a special command.

**Review by tomusdrw — CHANGES_REQUESTED** [review]

File: tools/links-check/index.ts [hunk 4]
```diff
@@ -7,31 +12,93 @@ main().catch((err: unknown) => {
...
+    } catch (err) {
+      const parentPath = path.dirname(currentPath);
+      if (parentPath === currentPath) {
+        break;
+      }
+      currentPath = parentPath;
+    }
+  }
+  return null;
+}
```
[the remark this answers is not in the document]
Comment (skoszuta) [reply tools/links-check/index.ts:31]: all fair points. i did add a message to the output saying that we're using a gitignore file and which one but that doesn't solve other issues.

File: tools/links-check/migrate.ts (hunk 3 above)
Comment (skoszuta) [reply tools/links-check/migrate.ts]: removed, will appear in a separate PR

**Review by tomusdrw — APPROVED** [review]

**Pull request merged**